### DISCUSSION TITLE
OptiFarm App Ideas Discussions - Features, Changes

### DISCUSSION THREAD

**Discussion by renwid** [OptiFarm/optiFarmMobileExpo#44] [Ideas]
## 👋 Welcome!
  We can use this tab to outline the list of upcoming features/changes that we would like to see inside the App

**Comment by renwid** [comment]
Niall mentioned that we could have our horizontal scroll associated with colours, e.g. the herd book could be a blue, medication usage could be yellow, etc.

**Comment by renwid** [comment]
Block could be made smaller because of the unused space, or we could include other information such as the herd size or come up with a synopsis of the herd book for this area.

**Comment by renwid** [comment]
Compare the expiry date to the current date and when it is getting close or has crossed the date then send notification to the user.

**Comment by renwid** [comment]
Add placeholders to the forms, giving intelligent assistance to the user so that they don't enter the wrong data.

**Reply by renwid** [reply]
Issue #64

**Comment by renwid** [comment]
Break into blocks, so you would have the date related fields together, then another section for animal related info, etc...

Possibly we could split it into different form pages, so at the bottom you would say, "Page 1 of 2" or "Page 2 of 2" or something like that if the page was split.

**Comment by renwid** [comment]
Alternate the colour scheme of the blocks, e.g. medication blocks are all the same colour, we could change this to grey, blue, grey, blue, ...., just to break up the scroll monotony.

**Comment by conorclarke27** [comment]
If the animal is female, you should be able to see the date when that animal last gave birth

**Reply by AshleyArmstrongS** [reply]
Additional field to animal, only display if not null, last calved. Field would be filled in by API no extra functionality on front-end other than displaying it. Just updating this based on what we discussed with Joe

**Comment by conorclarke27** [comment]
Enda's idea for feature: voice button records farmer's comments for a few seconds, small sized file stored in database

**Reply by conorclarke27** [reply]
Possible testing using two users:

Primary user: person administering the medication
Secondary user: person who records the information

**Comment by AshleyArmstrongS** [comment]
New enum field for medicine, type? vaccination, dose, antibiotic.

**Reply by conorclarke27** [reply]
Important for farmer to be aware of **what is and what isn't** an _antibiotic_. They have to use it all and not keep it stored for a long period of time. 

Primary reason to add this is to just get farmer to stop and think about whether the medication is an antibiotic or not. Caution is always advised when handling medication!!!

**Comment by conorclarke27** [comment]
Joe (mentioned in meeting) - weather data might be important but not essential for the user, possible future feature.

**Comment by conorclarke27** [comment]
Sorting animals - 
sort animals by last animal treated
sort medication by used the most (mastitus tubes for dairy farmers)

**Comment by conorclarke27** [comment]
Group dosing, mentioned by Joe, but after some deliberation not feasible within our time-frame. Might be a good feature for the future.

**Comment by conorclarke27** [comment]
View Progeny screen:

shows offspring of that animal, so you can see when they calved, if they are calving regularly or if the breeding cycle is not great (trouble with animal, maybe have to sell them), etc...

**Comment by conorclarke27** [comment]
For future release for testing (advice from Frank):

give user group task to perform - single e's question or system usability 

very specific tasks to do - feedback comes back as a score

linking derek(persona) to the demo...

**Comment by conorclarke27** [comment]
Add the ability to give medication from the My Herd page (mentioned by user during testing), possibly drag from right/left to bring button up.
